### ISSUE TITLE
[Accessibility]: Search results ARC checklist issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34738]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

- [ ] On empty search results page, toggles (case sensitivity, etc) in "The search bar" section and other example searches should be aria-hidden
![CleanShot 2022-04-29 at 16 26 23](https://user-images.githubusercontent.com/206864/166080326-400e8be8-e6d2-4bda-ac81-bf46201795b0.png)
- [ ] Search result type icon is not labelled (should be labelled based on type of result)
![CleanShot 2022-04-29 at 16 27 42](https://user-images.githubusercontent.com/206864/166080398-2e819683-da1a-4e1d-9bae-7ee6fb49709c.png)
![CleanShot 2022-04-29 at 16 28 10](https://user-images.githubusercontent.com/206864/166080447-820e0b74-1a1c-4241-a24f-703c3a2088d4.png)
- [ ] Code host icon should have role="img" and label on it directly instead of nested inside a span
![CleanShot 2022-04-29 at 16 28 41](https://user-images.githubusercontent.com/206864/166080480-b2e3a3c0-cba4-4a35-ba0b-4e20fe22f56a.png)
- [ ] Star icon is not labelled (should be labelled "n stars", where n is the number of stars; the number next to it should be aria-hidden)
![CleanShot 2022-04-29 at 16 29 12](https://user-images.githubusercontent.com/206864/166080513-78dcc9ad-d111-4cbb-b585-45291e58dcb1.png)
- [ ] Last sync icon is not labelled (should have label same as the tooltip content
![CleanShot 2022-04-29 at 16 29 31](https://user-images.githubusercontent.com/206864/166080539-877c0cf9-f50f-43fa-9284-8f66d2eef587.png)

### Expected behavior

All issues should be fixed as described above

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by gitstart-sourcegraph** [comment]
Hi @limitedmage 
want to clarify this requirement `On empty search results page, toggles (case sensitivity, etc) in "The search bar" section and other example searches should be aria-hidden`
specifically this part `other example searches should be aria-hidden`, would you mind sharing a little more detail, to have enough context. Thanks
